libs/util/httpRequest.js

```diff
@@ -7,49 +7,36 @@
  * @module httpRequest
  */
 
-function FetchrError(options, request, response, responseBody, originalError) {
-    var err = originalError;
-    var status = response ? response.status : 0;
-    var errMessage, errBody;
-
-    if (!err && (status === 0 || (status >= 400 && status < 600))) {
-        if (typeof responseBody === 'string') {
-            try {
-                errBody = JSON.parse(responseBody);
-                if (errBody.message) {
-                    errMessage = errBody.message;
-                } else {
-                    errMessage = responseBody;
-                }
-            } catch (e) {
-                errMessage = responseBody;
-            }
-        } else {
-            errMessage = status
-                ? 'Error ' + status
-                : 'Internal Fetchr XMLHttpRequest Error';
-        }
-
-        err = new Error(errMessage);
-        err.body = errBody || responseBody;
-        if (err.body) {
-            err.output = err.body.output;
-            err.meta = err.body.meta;
-        }
-    }
-
-    err.rawRequest = {
+function FetchrError(message, options, request, response) {
+    this.body = null;
+    this.message = message;
+    this.meta = null;
+    this.name = 'FetchrError';
+    this.output = null;
+    this.rawRequest = {
         headers: options.headers,
         method: request.method,
         url: request.url,
     };
-    err.statusCode = status;
-    err.timeout = options.timeout;
-    err.url = request.url;
-
-    return err;
+    this.statusCode = response ? response.status : 0;
+    this.timeout = options.timeout;
+    this.url = request.url;
+
+    if (response) {
+        try {
+            this.body = JSON.parse(message);
+            this.output = this.body.output || null;
+            this.meta = this.body.meta || null;
+            this.message = this.body.message || message;
+        } catch (e) {
+            this.body = message;
+        }
+    }
 }
 
+FetchrError.prototype = Object.create(Error.prototype);
```

Comment:
Why not make it a proper class?

Reply:
To keep es5 compatibility. I will transform it as soon as we move to TS, but for now, I think it's better to keep es5 syntax compatibility.